torch/_tensor_str.py

```diff
@@ -145,7 +145,11 @@ def format(self, value):
                 ret = ('{{:.{}f}}').format(PRINT_OPTS.precision).format(value)
         elif self.complex_dtype:
             p = PRINT_OPTS.precision
-            ret = '({{:.{}f}} {{}} {{:.{}f}}j)'.format(p, p).format(value.real, '+-'[value.imag < 0], abs(value.imag))
+            # format real and imaginary values according to type
+            if not is_float_tensor:
```

Comment:
what are you trying to do here?

Strictly speaking, is_floating_point() returns false for complex, so it's misleading to have is_float_tensor check and do something for it under complex branch

Reply:
You said we only want to truncate the zeros only in cases where all elements of the complex are integers. This checks to see if the complex tensor has any float components. I can change the variable name, but that's what it does.

Reply:
hmm maybe change it to has_non_zero_decimal_val ?

Reply:
Oh, I changed it to ```complex_contains_float```.  Can change it to```has_non_zero_decimal_val ``` if you prefer that though, let me know!

Reply:
yeah let's change it to ```has_non_zero_decimal_val``` :)

Reply:
Okay! Doing it right now.

Reply:
Updated PR, with values consistent as shown in the PR description.